ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+            }
+            else
+            {
+                if (i_reg == 0)
+                    break;
+                i_reg--;
+            }
+        }
+
+        if (whites)
+            free_function(whites);
+        free_function(record);
+
+        retorn_TruncaFitxer = TruncateFile_function(
+            data_base_XP->pfDataBase,
+            (MM_FILE_OFFSET)data_base_XP->FirstRecordOffset +
+                (MM_FILE_OFFSET)data_base_XP->nRecords *
+                    ((MM_FILE_OFFSET)data_base_XP->BytesPerRecord +
+                     canvi_amplada));
+        if (canvi_amplada < 0 && retorn_TruncaFitxer)
+            return 1;
+    } /* Fi de registres de != 0*/
+
+    if (canvi_amplada != 0)
+    {
+        data_base_XP->pField[nIField].BytesPerField = nNewWidth;
+        data_base_XP->BytesPerRecord += canvi_amplada;
+        for (i_camp = (MM_EXT_DBF_N_FIELDS)(nIField + 1);
+             i_camp < data_base_XP->nFields; i_camp++)
+            data_base_XP->pField[i_camp].AcumulatedBytes += canvi_amplada;
+    }
+    data_base_XP->pField[nIField].DecimalsIfFloat = nNewPrecision;
+
+    //DonaData(&(data_base_XP->day), &(data_base_XP->month), &(data_base_XP->year));
+
+    if ((MM_UpdateEntireHeader(data_base_XP)) == FALSE)
+        return 1;
+
+    return 0;
+} /* End of MMChangeCFieldWidthDBF() */
+
+static void MM_AdoptHeight(double *desti, const double *proposta, uint32_t flag)
+{
+    if (*proposta == MM_NODATA_COORD_Z)
+        return;
+
+    if (flag & MM_STRING_HIGHEST_ALTITUDE)
+    {
+        if (*desti == MM_NODATA_COORD_Z || *desti < *proposta)
+            *desti = *proposta;
+    }
+    else if (flag & MM_STRING_LOWEST_ALTITUDE)
+    {
+        if (*desti == MM_NODATA_COORD_Z || *desti > *proposta)
+            *desti = *proposta;
+    }
+    else
+    {
+        // First coordinate of this vertice
+        if (*desti == MM_NODATA_COORD_Z)
+            *desti = *proposta;
+    }
+}
+
+int MM_GetArcHeights(double *coord_z, FILE_TYPE *pF, MM_N_VERTICES_TYPE n_vrt,
+                     struct MM_ZD *pZDescription, uint32_t flag)
+{
+    MM_N_HEIGHT_TYPE i;
+    MM_N_VERTICES_TYPE i_vrt;
+    double *pcoord_z;
+    MM_N_HEIGHT_TYPE n_alcada, n_h_total;
+    int tipus;
+    double *alcada = nullptr, *palcada, *palcada_i;
+#define MM_N_ALCADA_LOCAL 50  // Nr of local heights
+    double local_CinquantaAlcades[MM_N_ALCADA_LOCAL];
+
+    for (i_vrt = 0; i_vrt < n_vrt; i_vrt++)
+        coord_z[i_vrt] = MM_NODATA_COORD_Z;
+
+    tipus = MM_ARC_HEIGHT_TYPE(pZDescription->nZCount);
+    n_alcada = MM_ARC_N_HEIGHTS(pZDescription->nZCount);
+    if (n_vrt == 0 || n_alcada == 0)
+        return 0;
+
+    if (tipus == MM_ARC_HEIGHT_FOR_EACH_VERTEX)
+        n_h_total = (MM_N_HEIGHT_TYPE)n_vrt * n_alcada;
+    else
+        n_h_total = n_alcada;
+
+    if (n_h_total <= MM_N_ALCADA_LOCAL)
+        palcada = local_CinquantaAlcades;
+    else
+    {
+        if (MMCheckSize_t(n_vrt * sizeof(double) * n_alcada, 1))
```

Comment:
Assuming overflow of n_vrt * n_alcada has been checked above, then below would be more efficient to detect overflow when multiplying that by sizeof(double)
```suggestion
        if (MMCheckSize_t(n_vrt * n_alcada, sizeof(double)))
```

Reply:
or maybe ?
```suggestion
        if (MMCheckSize_t(n_h_total, sizeof(double)))
```

Reply:
I think you did it by yourself. So I close the comment